ecl/hql/hqlgram2.cpp

```diff
@@ -11250,6 +11256,19 @@ bool HqlGram::isSaved(IHqlExpression * failure)
     return false;
 }
 
+bool HqlGram::isCritical(IHqlExpression * failure)
+{
+    if (!failure) return false;
```

Comment:
minor: This could have been coded as

    queryOperatorInList(no_critical, failure) != nullptr;